packages/block-editor/src/hooks/border.js

```diff
@@ -35,6 +34,13 @@ export function BorderPanel( props ) {
 	const isWidthSupported =
 		useSetting( 'border.width' ) && hasBorderSupport( props.name, 'width' );
 
+	const isDisabled = [
+		! isColorSupported,
+		! isRadiusSupported,
+		! isStyleSupported,
+		! isWidthSupported,
+	].every( Boolean );
+
 	if ( isDisabled || ! isSupported ) {
```

Comment:
As per the previous comment, `isDisabled` wraps up support checks as well now.

```suggestion
	if ( isDisabled ) {
```